src/main/java/com/example/videotecha/model/controller/AuthenticationController.java

```diff
@@ -0,0 +1,27 @@
+package com.example.videotecha.model.controller;
+
+import com.example.videotecha.dto.UserDto;
+import com.example.videotecha.model.RegisteredUser;
+import com.example.videotecha.service.AuthenticationService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(value = "/auth", produces = MediaType.APPLICATION_JSON_VALUE)
+public class AuthenticationController {
+
+    AuthenticationService authenticationService;
+
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<RegisteredUser> register(@RequestBody UserDto user) {
+        RegisteredUser registeredUser = authenticationService.register(user);
+        return new ResponseEntity<>(registeredUser, HttpStatus.OK);
+    }
```

Comment:
No need to return `@ResponseEntity` if the controller is annotated with `@RestController`. 

https://www.baeldung.com/spring-controller-vs-restcontroller